test/integration/tcp_proxy_integration_test.cc

```diff
@@ -104,6 +104,11 @@ TEST_P(TcpProxyIntegrationTest, TcpProxyUpstreamWritesFirst) {
   // Any time an associated connection is destroyed, it increments both counters.
   test_server_->waitForCounterGe("cluster.cluster_0.upstream_cx_destroy", 1);
   test_server_->waitForCounterGe("cluster.cluster_0.upstream_cx_destroy_with_active_rq", 1);
+
+  IntegrationTcpClientPtr tcp_client2 = makeTcpConnection(lookupPort("tcp_proxy"));
```

Comment:
What is this new test code covering?

Reply:
a reconnect corner case which I thought might be the issue (it wasn't ;-( )